include/helper_functions.php

```diff
@@ -1,15 +1,169 @@
+    $temp_array = [];  
+    $i = 0; 
+    $key_array = []; 
+    
+    foreach($array as $val) { 
+        if (!in_array($val[$key], $key_array)) {                                                                      //If the key is not already in the temp array stored it
+            $key_array[$i] = $val[$key]; 
+            $temp_array[$i] = $val; 
+        } 
+        $i++; 
+    } 
+    return $temp_array; 
+}
+
+// --------------------------------------------------------------------------
+
+function calculateDaysToNextClass($dates){
+    $today = new DATETIME(currentDate());
+    $days = [];
+    $i = 0;
+
+    foreach($dates as $date){        
+        $classDate = new DATETIME($date['Date']);      
+        $days[$i] = getDayDifference($classDate, $today);
+        $i++;       
+    }
+   
+    return $days;    
+}
+
+// --------------------------------------------------------------------------
+
+function calcAllRemainingCredits($allStudents, $allCredits, $allClasses){         
+
+    global $remainingCredits;
+
+    for($i = 0; $i < sizeof($allStudents); $i++){       
+
+        if($allCredits[$i]['Student_Id'] == $allStudents[$i]['Student_Id'] &&                       //Check if the function above did the calculations
+           $allClasses[$i]['Student_Id'] == $allStudents[$i]['Student_Id']){
+                                                                                                    //Might need to check something before?
+            $tempAmount = $allCredits[$i]['Credits_Amount'] - $allClasses[$i]['Classes_Amount'];    //Subtract the credits total by the classes total 
+                                                                                                    //This caculation is done based on the constraint that each class booked is 60 minutes (not more)
+            $remainingCredits[$i]['Student_Id'] = $allStudents[$i]['Student_Id'];                     
+            $remainingCredits[$i]['Remaining_Credits'] =  $tempAmount;
+        }
+
+        else{
+            echo"Student Id does not match";                                                       
+        }                                                                                           //Eventually this will be a different error message or something
+    } 
+}
+
+// --------------------------------------------------------------------------
+
+function calcStudentRemainingCredits($studentId){
+    $remainingCredits = [];  
+    $remainingCredits = getStudentCreditAmount($studentId);
+  
+    (!$remainingCredits)? $remainingCredits = '0': $remainingCredits = getStudentCreditAmount($studentId)['Credits_Amount'] - getStudentClassesAmount($studentId)['Classes_Amount'];
+    
+    return $remainingCredits;
+}
+
+// --------------------------------------------------------------------------
+
+function currentDate(){
+    date_default_timezone_set("America/Chicago");
```

Comment:
I'd run this in `initialize.php` so it's always set correctly (otherwise you risk forgetting to do this).

Once you've done that, i'm not sure you really need a function for this (it's not much different from just calling `date('Y/m/d h:i:s')